pipeline/matcher/scripts/getMatcherGraph.ts

```diff
@@ -0,0 +1,191 @@
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import fetch from 'node-fetch';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocument, QueryCommandInput } from '@aws-sdk/lib-dynamodb';
+import { getCreds } from '@weco/ts-aws/sts';
+import { digraph, INode, RootCluster, toDot } from 'ts-graphviz';
+import { exec } from 'child_process';
+import { SourceWork } from './models';
+import { getAttributes } from './graphAttributes';
+
+type UserInput = {
+  canonicalId: string;
+  pipelineDate: string;
+}
+
+type ElasticConfig = {
+  worksIndex: string;
+  imagesIndex: string;
+}
+
+
+async function getInput(): Promise<UserInput> {
+  let id: string;
+  if (process.argv.length != 3) {
+    console.error('Usage: getGraph <WORK_ID>');
+    process.exit(1);
+  } else {
+    id = process.argv[2];
+  }
+
+  const resp = await fetch('https://api.wellcomecollection.org/catalogue/v2/_elasticConfig');
+
+  if (resp.status !== 200) {
+    throw Error('Could not fetch ElasticConfig from API');
+  }
+
+  const elasticConfig: ElasticConfig = await resp.json();
+  const pipelineDate = elasticConfig.worksIndex.replace("works-indexed-", "");
+
+  return {
+    canonicalId: id,
+    pipelineDate: pipelineDate,
+  };
+}
+
+async function queryAllMatchingItems(client: DynamoDBDocument, query: QueryCommandInput): Promise<Array<Record<string, any>>> {
+  let allItems = [];
+
+  while (true) {
+    const result = await client.query(query);
+
+    if (!result.Items || result.Items.length == 0) {
+      break;
+    }
+
+    if (result.Items && result.Items.length > 0) {
+      allItems = [...allItems, ...result.Items]    
+    }
+
+    if (typeof result.LastEvaluatedKey === 'undefined') {
+      break;
+    }
+
+    query.ExclusiveStartKey = result.LastEvaluatedKey;
```

Comment:
Can we do this without mutating arguments? We could clone the `query` or - my preference - this function could be implemented recursively